src/components/Patient/PatientRegistration.tsx

```diff
@@ -62,63 +137,19 @@ export default function PatientRegistration(
   const { t } = useTranslation();
   const { goBack } = useAppHistory();
 
-  const [samePhoneNumber, setSamePhoneNumber] = useState(false);
-  const [sameAddress, setSameAddress] = useState(true);
-  const [ageDob, setAgeDob] = useState<"dob" | "age">("dob");
-  const [_showAutoFilledPincode, setShowAutoFilledPincode] = useState(false);
-  const [form, setForm] = useState<Partial<PatientModel>>({
-    nationality: "India",
-    phone_number: phone_number || "+91",
-    emergency_phone_number: "+91",
-  });
-  const [feErrors, setFeErrors] = useState<
-    Partial<Record<keyof PatientModel, string[]>>
-  >({});
   const [suppressDuplicateWarning, setSuppressDuplicateWarning] =
     useState(!!patientId);
   const [debouncedNumber, setDebouncedNumber] = useState<string>();
 
-  const sidebarItems = [
-    { label: t("patient__general-info"), id: "general-info" },
-    // { label: t("social_profile"), id: "social-profile" },
-    //{ label: t("volunteer_contact"), id: "volunteer-contact" },
-    //{ label: t("patient__insurance-details"), id: "insurance-details" },
-  ];
-
-  const mutationFields: (keyof PatientModel)[] = [
-    "name",
-    "phone_number",
-    "emergency_phone_number",
-    "geo_organization",
-    "gender",
-    "blood_group",
-    "date_of_birth",
-    "age",
-    "address",
-    "permanent_address",
-    "pincode",
-    "nationality",
-    "meta_info",
-    "ration_card_category",
-  ];
-
-  const mutationData: Partial<PatientModel> = {
-    ...Object.fromEntries(
-      Object.entries(form).filter(([key]) =>
-        mutationFields.includes(key as keyof PatientModel),
-      ),
-    ),
-    date_of_birth:
-      ageDob === "dob" ? dateQueryString(form.date_of_birth) : undefined,
-    age: ageDob === "age" ? form.age : undefined,
-    meta_info: {
-      ...(form.meta_info as any),
-      occupation:
-        form.meta_info?.occupation === ""
-          ? undefined
-          : form.meta_info?.occupation,
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      nationality: "India",
+      phone_number: phone_number || "+91",
+      emergency_phone_number: "+91",
+      yob_or_dob: "dob",
     },
-  };
+  });
 
   const createPatientMutation = useMutation({
```

Comment:
although outside the diff range, this is the convention we follow in most of the places.

```suggestion
  const { mutate: updatePatient } = useMutation({
```